### ISSUE TITLE
App doesn't close because of getLatestVersion failing and taking a long time to close the thread

### ISSUE THREAD

**Issue by jrainville** [status-im/status-desktop#4221]
This problem is two fold.

Right now, if we have a thread that has a task that takes a long time to finish and we close the app while it is running, the app won't fully close until that thread is done.
That could be fixed by finding a way to send info to that thread that we are closing and then that thread could stop it's operation, but I don't know if it's doable easily, especially with an HTTP request that is blocking as will be explained next.

The other part  of the problem is that `getLatestVersion` is currently broken and the IPFS files are unavailable because they were not pinned. To fix this, we need to re-upload the files, pin them and point the `desktop.status.eth` address to the IPFS hash. Richard is the owner of the domain for now.

So, to fix this issue, it would require to either fix one or both of those problems.

Logs:

```
ERR 2021-12-01 10:37:51.717+03:00 Error while getting latest version information tid=5407285 file=view.nim:147 msg="408 Request Timeout" Traceback (most recent call last) /Users/jenkins/workspace/s-desktop_branches_macos_PR-4194/src/app_service/tasks/threadpool.nim(113) taskThread /Users/jenkins/workspace/s-desktop_branches_macos_PR-4194/vendor/status-lib/vendor/nim-task-runner/vendor/nim-chronos/chronos/asyncloop.nim(1082) waitFor /Users/jenkins/workspace/s-desktop_branches_macos_PR-4194/vendor/status-lib/vendor/nim-task-runner/vendor/nim-chronos/chronos/asyncloop.nim(279) poll /Users/jenkins/workspace/s-desktop_branches_macos_PR-4194/src/app_service/tasks/threadpool.nim(98) runner /Users/jenkins/workspace/s-desktop_branches_macos_PR-4194/src/app/utilsView/view.nim(157) :anonymous /Users/jenkins/workspace/s-desktop_branches_macos_PR-4194/src/app_service/tasks/qt.nim(16) finish /Users/jenkins/workspace/s-desktop_branches_macos_PR-4194/vendor/nimqml/src/nimqml.nim(42) signal_handler SIGSEGV: Illegal storage access. (Attempt to read from nil?) QObject::~QObject: Timers cannot be stopped from another thread
```

**Comment by anastasiyaig** [comment]
@iurimatias this is the issue about logout

**Comment by PascalPrecht** [comment]
Thanks for detailing this issue!

> That could be fixed by finding a way to send info to that thread that we are closing and then that thread could stop it's operation, but I don't know if it's doable easily, especially with an HTTP request that is blocking as will be explained next.

I think this really boils down to the task runner having an API to close threads Whether there's an HTTP call or or any other I/O happening behind it shouldn't matter too much (although,  the in-flight HTTP request can't be cancelled anymore, but that shouldn't keep us from closing the thread).

I've looked into the code and was surprised we're checking for updates on initialization right away, but I guess it does make sense if we want to prompt users that a new version is available.
One thing we can do for now (at least to have it unblock the release), is to **not** check for updates implicitly, but wait for users to actually perform the check. 

We would then at least:

1. not have a broken sign out process anymore because the threaded task is pushed back to some later point in time. In fact it may never happen because the user might not run it (I guess though, the same issue will occur with any other thread task that takes too long, so this is really just fighting symptoms)
1.1 Another option would be to see if we could put a timeout into the HTTP call so that even if it replies with a bad gateway (which takes time and is slow), we can at least control the amount of time we're willing to wait
2. make it explicit for the user in the sense that, when checks for updates are done, a loading indication can be shown (should do that either way btw). Then when the user closes the app beforehand, it's a bit more "expected" something might be wrong (instead looking at some freezing app).

> The other part of the problem is that getLatestVersion is currently broken and the IPFS files are unavailable because they were not pinned. To fix this, we need to re-upload the files, pin them and point the desktop.status.eth address to the IPFS hash. Richard is the owner of the domain for now.

Do we know what's the current process for uploading/pinning those files? Is this done manually by someone? Do we have to do this on a periodical basis, so it would make sense to automate this?

**Comment by jrainville** [comment]
>  Another option would be to see if we could put a timeout into the HTTP call so that even if it replies with a bad gateway

This would be very helpful. Right now, the `timeout` property of `httpclient` seems broken as listed here: https://github.com/nim-lang/Nim/issues/14807#issuecomment-921646701
Fixing that timeout would at least help in the case of http requests

> Do we know what's the current process for uploading/pinning those files? Is this done manually by someone? Do we have to do this on a periodical basis, so it would make sense to automate this?

Right now, @richard-ramos does it manually. Apparently, there is/was a tool that was in the works to make that process automatic.
Anyway, I feel like if we have to manually do that process each time we have a new release, it is not a good thing. We should check where the tool is at and see if in the meantime we don't want to do something else for the version check (like checking on Github or something)

**Comment by iurimatias** [comment]
no longer relevant and will be fixed by #4884
